src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -299,14 +311,18 @@ private static bool TryGetClassDeclarationList(INamedTypeSymbol typeSymbol, [Not
                         Debug.Assert(propertyNameNode != null);
 
                         SyntaxNode? propertyValueMode = childNodes.ElementAtOrDefault(1);
-                        if (propertyNameNode.Name.Identifier.ValueText == "TypeInfoPropertyName")
+                        string optionName = propertyNameNode.Name.Identifier.ValueText;
+
+                        if (optionName == nameof(JsonSerializableAttribute.TypeInfoPropertyName))
                         {
                             typeInfoPropertyName = propertyValueMode.GetFirstToken().ValueText;
                         }
-                        else
+                        else if (optionName == nameof(JsonSerializableAttribute.GenerationMode))
                         {
-                            Debug.Assert(propertyNameNode.Name.Identifier.ValueText == "GenerationMode");
-                            generationMode = (JsonSourceGenerationMode)Enum.Parse(typeof(JsonSourceGenerationMode), propertyValueMode.GetLastToken().ValueText);
+                            if (Enum.TryParse<JsonSourceGenerationMode>(propertyValueMode.GetLastToken().ValueText, out JsonSourceGenerationMode value))
+                            {
+                                generationMode = value;
+                            }
```

Comment:
Add "else throw" here? Or is ignoring fine?

Reply:
https://github.com/dotnet/runtime/pull/54527#discussion_r658373369